core/tx_list.go

```diff
@@ -366,17 +373,20 @@ func (l *txList) Add(tx *types.Transaction, priceBump uint64) (bool, *types.Tran
 		if cost := tx.Cost(); l.nativecostcap.Cmp(cost) < 0 {
 			l.nativecostcap = cost
 		}
-		if gasPrice := tx.GasPrice(); l.nativegaspricefloor == nil || l.nativegaspricefloor.Cmp(gasPrice) > 0 {
-			l.nativegaspricefloor = gasPrice
-		}
 	} else {
-		fee := tx.Fee()
+		var fee *big.Int = tx.Fee()
+		if tx.Type() == types.CeloDenominatedTxType {
+			var err error
+			fee, err = toCURRENCY(fee, feeCurrency, &txCtx)
+			if err != nil {
+				log.Error("Can't get rate for currency: ", "currency", feeCurrency.Hex())
```

Comment:
Let's preserve the original error whenever we can.
```suggestion
				log.Error("Can't get rate for currency: ", "currency", feeCurrency.Hex(), "err", err)
```

Reply:
Done